docs/viplab3.0/computation_result.md

```diff
@@ -9,62 +9,89 @@ and result objects, like stdout, stderr, images, links, etc.
 ## Example (informal)
 
 ```
-{ "computation" : "4598393-95bf-409a-98a5-ee375982c3e",  // uuid of corresponding computation
+{
+  "identifier" : "86165eea-14df-4a76-805a-09b21441cbf7",
+  "version" : "3.0.0"
+  "computation" : "4598393-95bf-409a-98a5-ee375982c3e",  // uuid of corresponding computation
   "status" : "final", // final and intermediate, 
   "timestamp" :   // creation time of this message in ISO-8601
-  "notifications" : { // injected user info object
-    "summary" : "(C chain)v1.9 failed.",
-    "elements" : [
+  "output" :
+    {
+      "stdout" : "", // base64url encoded content, mandatory, even if empty
+      "stderr" : "", // base64url encoded content, mandatory, even if empty
+    },
+  "artifacts" : [ // additional artifacts (files, notifications, etc)
+    {
+      "type": "notifications", // each artifact requires a type and
+      "identifier" : "1dd479d9-c9bd-4711-aed8-5bfb3ec5fcfa", // and an identifier
+      "summary" : "(C chain)v1.9 failed.",
+      "elements" : [
         {
-            "severity" : "info",
-            "type" : "compiler",
-            "message" : "source_0.c: In function \u2018trapez\u2019:",
-            "output" : {
-                "elementID" : "compile_stderr_0.txt",
-                "extract" : "source_0.c: In function \u2018trapez\u2019:",
-                "begin" : 0,
-                "end" : 37
-            }
+          "severity" : "info",
+          "type" : "compiler",
+          "message" : "source_0.c: In function \u2018trapez\u2019:",
+          "output" : {
+            "elementID" : "compile_stderr_0.txt",
+            "extract" : "source_0.c: In function \u2018trapez\u2019:",
+            "begin" : 0,
+            "end" : 37
+          }
```

Comment:
"elements" and "elementID" are from the old ViPLab format and should be replaced. We should reevaluate the `notifications` artifact type. Maybe each notification is a seperate artifact?

Reply:
I did fix the the elementID. I did keep the elements array for now. This allows grouping of notifications together. E.g. all compiler warnings can go in one notifications object. I would leave it to the corresponding backend how it decides to group it messages. The client should be able to deal with several notifications artifacts.